src/main/java/com/beowulfe/hap/accessories/characteristics/Color.java

```diff
@@ -1,16 +1,15 @@
-package com.beowulfe.hap.accessories;
+package com.beowulfe.hap.accessories.characteristics;
 
 import com.beowulfe.hap.HomekitCharacteristicChangeCallback;
 import java.util.concurrent.CompletableFuture;
 
 /**
- * Extends {@link Lightbulb} with color settings. This will usually be implemented along with {@link
- * DimmableLightbulb}, but not necessarily so.
+ * Extends a Lightbulb with color settings. This will usually be implemented along with {@link
+ * Brightness}, but not necessarily so.
  *
  * @author Andy Lintner
  */
-public interface ColorfulLightbulb extends Lightbulb {
-
+public interface Color {
```

Comment:
I like where this is going. After some thought, I wonder if this interface is probably more trouble than it's worth.

Perhaps we can just axe it and have the respective characteristics take lambda functions as appropriate. For read-only characteristics, it takes only a getter function and a subscription. For write-only, only a setter. Then the respective accessories can just return an Optional of the characteristic, with the lambda functions defined as needed?

WDYT?

Reply:
Yup, I like that.